static/styles/compose.css

```diff
@@ -305,7 +305,12 @@ textarea.new_message_textarea {
     width: calc(100% - 12px);
     padding: 5px;
     height: 1.5em;
-    max-height: 22em;
+
+    /* 50vh will ensure at most half of the viewport is taken by the
+    compose box at max and the 50px is for the top header that we have.
+    Thus, always have an ample amount of space to scroll
+    Issue #16038 */
+    max-height: calc(50vh - 50px);
```

Comment:
Can you paste before/after some screenshots to support this change with different screen sizes?

Reply:
In case of this change, when the height is less, it ensures some space is left above.
![image](https://user-images.githubusercontent.com/56730716/122197313-af46fe80-ceb5-11eb-8cd9-332e83566aff.png)

whereas before it was `22em`
![image](https://user-images.githubusercontent.com/56730716/122197506-e1f0f700-ceb5-11eb-8fcc-ceddb8d3e2c4.png)

These screenshots are taken with max height possible.

In case of devices having some height,
Now
![image](https://user-images.githubusercontent.com/56730716/122197824-34caae80-ceb6-11eb-8026-191931cbd23e.png)

Before it was
![image](https://user-images.githubusercontent.com/56730716/122197710-1a90d080-ceb6-11eb-8a75-1d8f7e30125d.png)

Reply:
Let's drop this commit given #16038; I think we can consider changing the default height of `bottom_whitespace` separately, but there's no need to change it in this PR.

Reply:
Dropped.